### PR TITLE
Add methods to `ProtoWorld` for working with `BlockState`s

### PR DESCRIPTION
**Pull request by dfsek**
I had a cosmic brain moment in #5727 and forgot to implement access to `BlockState`. This PR corrects this.

I've updated the test generator to test the new API as well: https://gist.github.com/dfsek/4b79e12f19fd3574a32701871a3aa8ed

image
image

### CODE REVIEW

File: patches/server/0714-Add-Feature-Generation-API.patch [hunk 1]
```diff
@@ -121,3 +142,517 @@ index 46ce973ca68420825316352ad4df907edf52bfec..65e9ecda0df8a22f93f5979a8c9ac74e
...
+-            }
+-            return new CraftBeehive(this.material, (BeehiveBlockEntity) te);
+-        case SCULK_SENSOR:
+-            if (te == null) {
+-                te = new SculkSensorBlockEntity(blockposition, iblockdata);
+-            }
+-            return new CraftSculkSensor(this.material, (SculkSensorBlockEntity) te);
+-        default:
+-            throw new IllegalStateException("Missing blockState for " + this.material);
++        switch (material) {
```
Comment (Proximyst) [inline patches/server/0714-Add-Feature-Generation-API.patch]: Don't touch indent here

**Review by Proximyst — CHANGES_REQUESTED** [review]
Looks fine

**Review by jpenilla — APPROVED** [review]
LGTM.

In the future please rebase instead of merging upstream for PRs, makes it a little easier when pulling

File: patches/api/0319-Add-Feature-Stage-API.patch [hunk 2]
```diff
@@ -58,6 +59,48 @@ index 0000000000000000000000000000000000000000..cf9443f1cd3039f0e53c645c89448302
 +    }
 +
 +    /**
++     * Sets the BlockState at a location.
```
Comment (MiniDigger) [inline patches/api/0319-Add-Feature-Stage-API.patch]: do we @Link here normally? if so, adjust the others below too

File: patches/server/0714-Add-Feature-Generation-API.patch [hunk 3]
```diff
@@ -121,3 +142,257 @@ index 46ce973ca68420825316352ad4df907edf52bfec..65e9ecda0df8a22f93f5979a8c9ac74e
...
++        return createBlockState(this.material, this.blockEntityTag);
++    }
++
++    public static BlockState createBlockState(Material material, CompoundTag blockEntityTag) {
++        Material stateMaterial = (material != Material.SHIELD) ? material : CraftMetaBlockState.shieldToBannerHack(blockEntityTag); // Only actually used for jigsaws
++        if (blockEntityTag != null) {
++            switch (material) {
+                 case SHIELD:
+-                    this.blockEntityTag.putString("id", "banner");
++                    blockEntityTag.putString("id", "banner");
```
Comment (MiniDigger) [inline patches/server/0714-Add-Feature-Generation-API.patch:168]: why have all these this qualifiers been nuked?

Comment (jpenilla) [reply patches/server/0714-Add-Feature-Generation-API.patch:168]: The method has been made static

**Review by Proximyst — APPROVED** [review]
LGTM. Merger can handle Mini's comment

**Pull request merged**